src/components/HeaderNotification.vue

```diff
@@ -94,42 +97,57 @@ export default {
       localListNotification: (state) => state.substrate.localListNotification
     })
   },
+
   props: {
     role: String
   },
+
   data: () => ({
     isLoading: false,
     balance: 0,
     name: "",
     listNotif: [],
     notifLength: 0,
-    showDialog: false
+    showDialog: false,
+    notifReads: []
   }),
+
   methods: {
     ...mapActions({
       clearAuth: "auth/clearAuth"
     }),
+
     openNotif() {
       if (this.showDialog) {
         this.showDialog = false;
       } else {
         this.showDialog = true;
       }
     },
+
     async getListNotification() {
       await this.$store.dispatch("substrate/getListNotification", {
         address: this.wallet.address,
         role: this.role
       });
       this.setData();
     },
+
     setData() {
       this.listNotif = this.localListNotification;
       this.notifLength = this.listNotif.filter((x) => x.read == false).length;
     },
-    gotoRoute(notif, index) {
+    
+    async gotoRoute(notif, index) {
       if (this.listNotif[index].read == false) {
+        clearTimeout(timeout)
         this.listNotif[index].read = true;
+        this.notifReads.push(notif.id)
+
+        timeout = setTimeout(async () => {
+          await setReadNotification(this.notifReads)
+        }, 2000)
```

Comment:
Why add a timeout?

Reply:
o debounce and reduce bulk request to API